source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp

```diff
@@ -493,49 +488,47 @@ bool SwiftASTManipulator::RewriteResult() {
       m_decl_context = decl_context;
     }
 
-    virtual bool walkToDeclPre(swift::Decl *decl) {
-      // Don't step into function declarations, they may have returns, but we
-      // don't want
-      // to instrument them.
+    bool walkToDeclPre(swift::Decl *D) override {
       swift::DeclKind kind = decl->getKind();
-      switch (kind) {
+      switch (D->getKind()) {
+      default: return true;
+
+      // Don't step into function declarations, they may have returns, but we
+      // don't want to instrument them.
       case swift::DeclKind::Func:
       case swift::DeclKind::Class:
       case swift::DeclKind::Struct:
         return false;
-      default:
-        return true;
       }
     }
 
-    virtual std::pair<bool, swift::Expr *> walkToExprPre(swift::Expr *expr) {
+    std::pair<bool, swift::Expr *> walkToExprPre(swift::Expr *expr) override {
+      switch (expr->getKind()) {
+      default: return {true, expr};
+
       // Don't step into closure definitions, they may have returns, but we
-      // don't want
-      // to instrument them either.
-      swift::ExprKind kind = expr->getKind();
-      if (kind == swift::ExprKind::Closure)
+      // don't want to instrument them either.
+      case swift::ExprKind::Closure:
         return {false, expr};
-      else
-        return {true, expr};
+      }
     }
 
-    virtual swift::Stmt *walkToStmtPost(swift::Stmt *stmt) {
-      swift::ReturnStmt *possible_return =
-          swift::dyn_cast<swift::ReturnStmt>(stmt);
-      if (possible_return && possible_return->hasResult()) {
-        swift::Expr *return_expr = possible_return->getResult();
-        if (return_expr) {
-          const bool add_return = true;
-          swift::Stmt *return_stmt;
+    swift::Stmt *walkToStmtPost(swift::Stmt *S) override {
+      auto *RS = swift::dyn_cast<swift::ReturnStmt>(S);
+      if (!RS)
+        return S;
 
-          return_stmt = m_manipulator.ConvertExpressionToTmpReturnVarAccess(
-              return_expr, possible_return->getStartLoc(), add_return,
-              m_decl_context);
-          if (return_stmt)
-            stmt = return_stmt;
+      if (RS->hasResult()) {
+        if (swift::Expr *RE = RS->getResult()) {
+          const bool add_return = true;
```

Comment:
Something I think looks nicer, feel free to ignore:

You could fold `true` into the call below and add a small comment like so:
`m_manipulator.ConvertExpressionToTmpReturnVarAccess(RE, RS->getStartLoc(), /*add_return*/ true, m_decl_context)`